### PR TITLE
Refactor internal messages/process.exit to support orderly exiting.

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 
+## 4.4.2 - 2021-10-xx
+
+### Fixed
+- Signal handlers that prematurely terminated the primary or its workers
+  have been refactored to allow an orderly exit. The primary and worker
+  internal messaging and exit process has been cleaned up and simplified to
+  help ensure more consistent outcomes and eliminate a number of bugs or
+  unexpected states. Additionally, some unused IPC messages were removed.
+  These changes should not have affected exiting applications except in those
```
Comment (davidlehn) [inline CHANGELOG.md:11]: Sounds odd to be talking about current changes in past tense.
```suggestion
  These changes should not affect exiting applications except in those
```

**Comment by dlongley** [comment]
This was manually tested with an external application, with 1-16 workers and sending SIGTERM and SIGINT to both workers and the primary process at various times during the application's life cycle. We'll want to see if we can engineer some unit tests for this in the future but it may be challenging.

File: CHANGELOG.md (hunk 1 above)
Comment (dlongley) [reply CHANGELOG.md:11]: Well, I wrote it that way because it was my intention not to affect -- but there could be bugs. I'll reword.

**Comment by dlongley** [comment]
Addresses #85.

**Review by gannan08 — APPROVED** [review]

**Comment by mattcollier** [comment]
Thanks @dlongley

It seems like this would be a good time to address this if possible:  https://github.com/digitalbazaar/bedrock/issues/80

Meanwhile, I can do some testing with this branch.

**Review by mattcollier — APPROVED** [review]

**Pull request merged**